catalog/ftp/src/main/java/ddf/catalog/ftp/ftplets/FtpRequestHandler.java

```diff
@@ -151,65 +232,118 @@ public FtpletResult onUploadStart(FtpSession session, FtpRequest request)
+                throw new FtpException("Failure to create metacard for file " + fileName);
+            } catch (Exception e) {
+                LOGGER.debug("Error getting the output data stream from the FTP session", e);
+                throw new IOException("Error getting the output stream from FTP session");
+            } finally {
+                session.getDataConnection()
+                        .closeDataConnection();
+            }
 
-                    if (LOGGER.isDebugEnabled()) {
-                        for (Metacard metacard : createdMetacards) {
-                            LOGGER.debug("Content item created with id = {}", metacard.getId());
-                        }
-                    }
-                } else {
-                    throw new FtpException();
-                }
-                return null;
-            });
-        } catch (FtpException fe) {
-            LOGGER.debug("Failure to create metacard for file {}", fileName, fe);
-            throw new FtpException("Failure to create metacard for file " + fileName);
-        } catch (Exception e) {
-            LOGGER.debug("Error getting the output data stream from the FTP session", e);
-            throw new IOException("Error getting the output stream from FTP session");
-        } finally {
-            session.getDataConnection()
-                    .closeDataConnection();
         }
 
         return FtpletResult.SKIP;
     }
 
-    protected String getMimeType(String fileExtension, FileBackedOutputStream outputStream) {
+    private CreateStorageRequest getCreateStorageRequest(String fileName,
+            TemporaryFileBackedOutputStream outputStream) throws IOException {
+        String fileExtension = FilenameUtils.getExtension(fileName);
+        String mimeType = getMimeType(fileExtension, outputStream);
+
+        ContentItem newItem = new ContentItemImpl(outputStream.asByteSource(),
+                mimeType,
+                fileName,
+                null);
+
+        return new CreateStorageRequestImpl(Collections.singletonList(newItem), null);
+    }
+
+    private List<Metacard> storeObject(Subject shiroSubject, String fileName,
+            CreateStorageRequest createRequest) {
+        return shiroSubject.execute(() -> {
+            CreateResponse createResponse = null;
+            try {
+                createResponse = catalogFramework.create(createRequest);
+            } catch (IngestException | SourceUnavailableException e) {
+                LOGGER.info("Failure to ingest file {}", fileName, e);
+            }
+
+            if (createResponse != null) {
+                if (LOGGER.isDebugEnabled()) {
+                    createResponse.getCreatedMetacards()
+                            .forEach(metacard -> LOGGER.debug("Content item created with id = {}",
+                                    metacard.getId()));
+                }
+            } else {
+                throw new FtpException();
```

Comment:
Seems like this would be more appropriate thrown above where the LOGGER.info is happening so it could wrap the `IngestException` or `SourceUnavailableException` with more details rather than just throwing an empty exception.